caddyhttp/httpserver/server.go

```diff
@@ -215,9 +215,20 @@ func (s *Server) Listen() (net.Listener, error) {
 		}
 	}
 
+	if tcpLn, ok := ln.(*net.TCPListener); ok {
+		ln = tcpKeepAliveListener{TCPListener: tcpLn}
+	}
```

Comment:
Pulled it out of Serve, since a TCPListener is not necessarily returned from Listen anymore.